tests/integration/functions/runtime_context/string_parameters_test.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  assert, IsExact
+} from "../../../../src/dev_deps.ts";
+import { CanBe, CanBeUndefined } from "../../../../src/test_utils.ts";
+import { DefineFunction } from "../../../../src/mod.ts";
+import {
+  EnrichedSlackFunctionHandler,
+} from "../../../../src/functions/types.ts";
+import { SlackFunctionTester } from "../../../../src/functions/tester/mod.ts";
+import { assertEqualsTypedValues } from "../../../../src/test_utils.ts";
+
+/**
+ * Custom Function handler tests, exercising string-based inputs and outputs
```

Comment:
Something that's interesting to me about these tests is that I feel like the intent of the tests is based on the functionality of the handler (defined inputs are available, defined outputs are expected), and less focused on the type of parameter. I think this is supported by the fact that we don't have tests for the other simple types (boolean, number, channel_id, etc.), but maybe we should.

Should we have this file focus on handler functionality like the inputs/outputs/optionality? Should we add tests for our other primitive types?

Reply:
I like focusing the tests on handler functionality like input, and output and optionality. As for how to organize them, some ideas, wondering what your thoughts are:

- could name the test files based on the type, and exercise inputs, outputs and optionality per type. So we'd keep the file name for this one as-is, but I could add a `boolean_parameters_test.ts`, `number_parameters_test.ts`, etc.
- could name the test files based on functionality (`input_parameters_test.ts`, `output_parameters_test`, `input_parameter_optionality_test.ts`) and exercise all primitive (and complex?) types in them
- could name the test files based on both! `string_input_parameters_test.ts`, `string_output_parameters.ts`, `string_input_parameter_optionality_test.ts`, etc. etc. Lotsa files, but presumably they would be tiny (maybe even only one or two tests per file).

Just some ideas off the top of my head. Lemme know what you think!

Reply:
I think let's start with option 2 — split into `input`, `output`, and `optionality`.

Then, either in this PR or another PR we could introduce type specific tests. I don't think we need to have exhaustive testing on types, just ensuring that at runtime a string is a string, a boolean is a boolean, etc.

Reply:
Done in 9ca9bff